arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -10,10 +10,13 @@
 #include <soc.h>
 #include <arch/arm/cortex_m/cmsis.h>
 #include <arch/arm/cortex_m/mpu/nxp_mpu.h>
-#include <logging/sys_log.h>
 #include <misc/__assert.h>
 #include <linker/linker-defs.h>
 
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(mpu);
```

Comment:
Another incarnation of problem mentioned earlier: The ARM driver registers the module and other MPU-related files uses this registration. However as ARM MPU driver is not used on NXP platforms, here you are trying to use unregistered module.